src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -208,6 +223,43 @@ public void UpdateAgentSetting(AgentSettings settings)
             settings.ProjectName = _projectName;
         }
 
+        public bool GetAddTagsRequired(CommandSettings command)
+        {
+            return command.GetAddMachineGroupTagsRequired();
+        }
+
+        public async Task GetAndAddTags(CommandSettings command, TaskAgent agent)
+        {
+            string tagString = command.GetMachineGroupTags();
+            Trace.Info("Given tags - {0} will be processed and added", tagString);
+
+            if (!string.IsNullOrWhiteSpace(tagString))
+            {
+                var tags =
+                    tagString.Split(',').Where(s => !string.IsNullOrWhiteSpace(s)).Select(s => s.Trim()).Distinct();
```

Comment:
You can use IgnoreCase flag in Distinct(). I mean, _Tag1_ and _tag1_ are same.
The DB system at the back of our service is CaseInsensitive. So, I don't think it will be turning case sensitive anytime soon.